cmd/tiller/release_server.go

```diff
@@ -349,6 +346,114 @@ func (s *releaseServer) prepareUpdate(req *services.UpdateReleaseRequest) (*rele
 	return currentRelease, updatedRelease, nil
 }
 
+func (s *releaseServer) RollbackRelease(c ctx.Context, req *services.RollbackReleaseRequest) (*services.RollbackReleaseResponse, error) {
+
+	currentRelease, targetRelease, err := s.prepareRollback(req)
+	if err != nil {
+		return nil, err
+	}
+
+	rel, err := s.performRollback(currentRelease, targetRelease, req)
+	if err != nil {
+		return nil, err
+	}
+
+	if err := s.env.Releases.Create(targetRelease); err != nil {
```

Comment:
@adamreese pointed out that if dry-rolling, this storage entry is still created causing dissonance between the reality and perception of history. Simple solution seems to be to just move this line to the end of performRollback.